packages/viewer/src/interactions/tapInteractionHandler.ts

```diff
@@ -148,25 +145,27 @@ export class TapInteractionHandler implements InteractionHandler {
 
   private handlePointerEnd(
     position?: Point.Point,
-    keyDetails: Partial<TapEventKeys> = {},
-    isTouch = false
+    keyDetails: Partial<TapEventKeys> = {}
   ): void {
+    const isDoubleTap =
+      this.doubleTapTimer != null && this.secondPointerDownPosition != null;
+    const isLongPress = this.longPressTimer == null;
+
     if (position != null && this.pointerDownPosition != null) {
-      if (
-        this.doubleTapTimer != null &&
-        this.secondPointerDownPosition != null
-      ) {
+      if (isDoubleTap) {
         this.emit(this.interactionApi?.doubleTap)(
           position,
           keyDetails,
+          { doubleTap: true, longPress: isLongPress },
           this.secondPointerDownPosition
         );
         this.clearDoubleTapTimer();
       }
 
-      if (this.longPressTimer != null) {
-        this.emit(this.interactionApi?.tap)(position, keyDetails);
-      }
+      this.emit(this.interactionApi?.tap)(position, keyDetails, {
+        doubleTap: isDoubleTap,
+        longPress: isLongPress,
+      });
```

Comment:
My experience from using most platforms is that click events are emitted before double click events.